### ISSUE TITLE
Unexpected memory consumption with apex.normalization.FusedLayerNorm

### ISSUE THREAD

**Issue by hyukyu** [NVIDIA/apex#928]
Hello,
I am facing a memory issue when using apex.normalization.FusedLayerNorm

When I call the forward function of `FusedLayerNorm`, my gpu:0 memory is consumed.
Below is the code that I've used for the test.

```import torch
from apex.normalization.fused_layer_norm import FusedLayerNorm

def main():
    layernorm = FusedLayerNorm(768, eps=1e-12).cuda(device=torch.device(1))
    embeddings = torch.randn((3, 102, 768), dtype=torch.float, device=torch.device(1))
    output = layernorm(embeddings)
    return output

if __name__ == "__main__":
    main()
```

And below is the memory status after the line `output = layernorm(embeddings)` is executed.
I am not sure why the process is eating up gpu:0 memory.

<img width="557" alt="Screen Shot 2020-08-04 at 4 27 16 PM" src="https://user-images.githubusercontent.com/19749157/89265407-62788180-d66f-11ea-921f-38b17414e1b3.png">

**Comment by benywon** [comment]
I have the same question, it seems that the mean and var of the apex.normalize.layernorm are calculated in CUDA0??
